src/components/InputImage.vue

```diff
@@ -1,74 +1,110 @@
+    :rules="validationRules"
+    v-slot="{validate}"
+    :bails="false"
+    class="d-block">
+      <div @click="launchFilePicker()">
+        <slot></slot>
+      </div>
+    <input
+      type="file"
+      ref="image"
+      :name="name"
+      @change="onFileChange($event.target.name, $event.target.files, validate)"
+      class="d-none">
+    <div class="mb-4"></div>
+    <transition-group tag="div" name="fade-transition">
+      <div v-for="(error, index) in validationErrors" :key="error">
+          <v-fade-transition>
+            <v-alert
+              :dense="alertDense"
+              :colored-border="alertColoredBorder"
+              :elevation="alertElevation"
+              :border="borderLocation"
+              :type="alertType"
+              dismissible
+              @input="deleteError(index)"
+              >
+              {{error}}
+            </v-alert>
+          </v-fade-transition>
+      </div>
+    </transition-group>
+  </ValidationProvider>
 </template>
 
 <script>
+import { ValidationProvider } from 'vee-validate'
+
 export default {
   name: 'InputImage',
+  components: {
+    ValidationProvider
+  },
+  props: {
+    name: {
+      type: String,
+      required: true
+    },
+    icon: {
+      type: String
+    },
+    value: {
+      required: true
+    },
+    validationRules: {
+      type: Object,
+      required: true
+    },
+    alertType: {
+      type: String
+    },
+    alertElevation: {
+      type: String
+    },
+    alertDense: {
+      type: Boolean
+    },
+    borderLocation: {
+      type: String
+    },
+    alertColoredBorder: {
+      type: Boolean
+    }
+  },
   data () {
     return {
-      errorDialog: null,
-      errorText: '',
-      uploadFieldName: 'image',
-      maxSize: 1024
+      validationErrors: []
     }
   },
-  props: {
-    value: Object
-  },
   methods: {
     launchFilePicker () {
       this.$refs.image.click()
     },
-    onFileChange (fieldName, file) {
-      const { maxSize } = this
-      let imageFile = file[0]
-      if (file.length > 0) {
-        let size = imageFile.size / maxSize / maxSize
-        if (!imageFile.type.match('image.*')) {
-          // check whether the upload is an image
-          this.errorDialog = true
-          this.errorText = 'Please choose an image file'
-        } else if (size > 1) {
-          // check whether the size is greater than the size limit
-          this.errorDialog = true
-          this.errorText = 'Your file is too big! Please select an image under 1MB'
-        } else {
-          // Append file into FormData and turn file into image URL
-          let formData = new FormData()
-          let imageURL = URL.createObjectURL(imageFile)
-          formData.append(fieldName, imageFile)
-          // Emit the FormData and image URL to the parent component
-          this.$emit('input', { formData, imageURL })
-        }
+    deleteError (index) {
+      this.$delete(this.validationErrors, index)
+    },
```

Comment:
Use Vuex mutations and actions.

Reply:
Again this is internal component method I think this should stay inside input image.

Reply:
I don't. Errors/infos can be global. Like you are currently offline.

Reply:
I have dilema, because using vuex in this case would mean that I can't use this component in the project without it. In addition to that I would have to store those errors in a way that allows me to delete errors on one specific input image component and the only way I can think of right now to do that is to identify them by name prop, but that would mean input names would need to be unique.

Reply:
So why do you use vuex as a project base? Missconception? What's use of it in your project?

Reply:
The thing is if you use vuex you don't have to use it in entire project. You suggested using it here, but I think using it here is not needed, because in future i would like to reuse this component in different projects. Tying this component to vuex will mean that i will need to include vuex in those projects if i want to use this component in it.